fili-core/src/main/java/com/yahoo/bard/webservice/table/ConcretePhysicalTable.java

```diff
@@ -26,19 +27,21 @@
      * @param timeGrain  time grain of the table
      * @param columns  The columns for this table
      * @param logicalToPhysicalColumnNames  Mappings from logical to physical names
+     * @param metadataService datasource metadata service containing availability data for the table
      */
     public ConcretePhysicalTable(
             @NotNull TableName name,
             @NotNull ZonedTimeGrain timeGrain,
-            @NotNull Iterable<Column> columns,
-            @NotNull Map<String, String> logicalToPhysicalColumnNames
+            @NotNull Set<Column> columns,
+            @NotNull Map<String, String> logicalToPhysicalColumnNames,
+            @NotNull DataSourceMetadataService metadataService
     ) {
         super(
                 name,
                 timeGrain,
                 columns,
                 logicalToPhysicalColumnNames,
-                new ImmutableAvailability(name, Collections.emptyMap())
+                new ConcreteAvailability(name, columns, metadataService)
```

Comment:
Do we want to be building the availability, or passing it in?

If we inject the `Availability` directly, it'll make it a lot easier for customers to provide their own custom availability rules...